x-pack/metricbeat/module/cloudfoundry/_meta/docs.asciidoc

```diff
@@ -0,0 +1,99 @@
+This is the cloudfoundry module.
+
+The Cloud Foundry module connects to Cloud Foundry loggregator to gather container, counter, and value metrics into a common data platform where it can be used for analysis, visualization, and alerting.
+
+
+The cloudfoundry module metrics are numerical values that describe some aspect of a system at a particular point in time. They are collected when pushed from the loggregator and are identified with a timestamp, a name, a value, and one or more defining labels.
+
+The cloudfoundry module mericsets are `container`, `counter` and `value`.
+
+[float]
+=== Module-specific configuration notes
+
+All metrics come from the Cloud Foundry loggregator API. The loggregator API authenticates through the Cloud Foundry UAA API.
+This requires that a new client be added to UAA with the correct permissions. This can be done using the `uaac` client.
+
+[source,bash]
+----
+$ export CLOUDFOUNDRY_CLIENT_ID=metricbeat
+$ export CLOUDFOUNDRY_CLIENT_SECRET=yoursecret
+$ uaac client add $CLOUDFOUNDRY_CLIENT_ID --name $CLOUDFOUNDRY_CLIENT_ID --secret $CLOUDFOUNDRY_CLIENT_SECRET --authorized_grant_types client_credentials,refresh_token --authorities doppler.firehose,cloud_controller.admin_read_only
+----
+
+Then configuration of the module needs to contain the created `client_id` and `client_secret`.
+
+[source,yaml]
+----
+- module: cloudfoundry
+  api_address: https://api.dev.cfdev.sh
+  client_id: "${CLOUDFOUNDRY_CLIENT_ID}"
+  client_secret: "${CLOUDFOUNDRY_CLIENT_SECRET}"
+  ssl:
+      verification_mode: none
+----
+
+
+[float]
+== Metricsets
+
+[float]
+=== `container`
+The container metricset of Cloud Foundry module allows you to collect container metrics that the
+loggregator sends to metricbeat.
+
+[float]
+=== `counter`
+The counter metricset of Cloud Foundry module allows you to collect counter metrics that the
+loggregator sends to metricbeat.
+
+[float]
+=== `value`
+The value metricset of Cloud Foundry module allows you to collect value metrics that the
+loggregator sends to metricbeat.
+
+
+[float]
+== Configuration options
+
+The `cloudfoundry` input supports the following configuration options.
+
+[float]
+=== `api_address`
+
+The URL of the Cloud Foundry API. Optional. Default: "http://api.bosh-lite.com".
+
+[float]
+=== `doppler_address`
```

Comment:
I think we need to put `doppler_address`, `uaa_address` and `rlp_address` into config.reference.yml so users know they have these options.

Reply:
Added